### DISCUSSION TITLE
Using formatters for a better code style?

### DISCUSSION THREAD

**Discussion by tb148** [ScratchAddons/ScratchAddons#21] [Discussion]
Discuss about which style should we use.
Choices:
- LLVM
- Google
- Chromium
- Mozilla
- Microsoft
- Webkit

I will work on it once the style is decided.

**Comment by WorldLanguages** [comment]
Err, what am I missing here? Clang? Does clang-format lint JavaScript?

**Comment by tb148** [comment]
> 
> 
> Err, what am I missing here? Clang? Does clang-format lint JavaScript?

YES! See [here](https://releases.llvm.org/10.0.0/tools/clang/docs/ClangFormat.html) and [here](https://releases.llvm.org/10.0.0/tools/clang/docs/ClangFormatStyleOptions.html) for clang-format-10, my current version.

**Comment by Explosion-Scratch** [comment]
Uhm, whatis clang?

**Comment by tb148** [comment]
> 
> 
> Uhm, whatis clang?

Check [here](https://releases.llvm.org/10.0.0/tools/clang/docs/ClangFormat.html)

**Comment by WorldLanguages** [comment]
I believe we should be using ESlint which is the standard library to lint JS. 
However, we should not require contributors to lint their code before merging their pull request so they don't need nothing more than a text editor.

**Comment by tb148** [comment]
> 
> 
> I believe we should be using ESlint which is the standard library to lint JS.
> However, we should not require contributors to lint their code before merging their pull request so they don't need nothing more than a text editor.

We can use github actions. Also this is called "formatting", not "linting".

**Comment by WorldLanguages** [comment]
Smart. Yeah sorry I don't know the difference tbh. I mean having consistent spacing between ifs and parenthesis, having a new line at the end of every file, etc. Unused variables or function parameters aren't really a problem.

**Comment by tb148** [comment]
> 
> 
> Smart. Yeah sorry I don't know the difference tbh. I mean having consistent spacing between ifs and parenthesis, having a new line at the end of every file, etc. Unused variables or function parameters aren't really a problem.

I will work on the github action. Can anyone recommend any formatter? If no one recommends any I will use clang-format.

**Comment by WorldLanguages** [comment]
I've used Prettier and it worked fine. I used it together with ESlint, I have no idea if they can go separated or not. If you can get it to work great! :)

**Comment by tb148** [comment]
I am very new to Prettier so I might not be able to use it in the github action. However I am still learning it!
